### PR TITLE
Fix #3185

### LINKED ISSUE #3185: `select()` with nested keys issue
When using the `select()` method with nested keys (i.e.: `key1.key2`) `key1` needs to be included automatically.

ex:

```
let query = new Parse.Query("MyObject");
query.select("key.key2");
// does not work
```

```
let query = new Parse.Query("MyObject");
query.include("key");
query.select("key.key2");
// works!
```

### PR DESCRIPTION
**Pull request by flovilmart**
When using `select() (i.e.: keys)` with nested keys `a.b` parse-server doesn't include by default `a` which make the call fail (unless you have `include('a')` specified).

This PR addresses that issue and will simplify the behaviour (and code for the clients).

For ex, calling select `a.b.c` will not include full objects `a` and `a.b` but only the necessary path to resolve correctly the `keys` clause.

Fixes #3185

### CODE REVIEW

File: src/RestQuery.js [hunk 1]
```diff
@@ -52,15 +53,36 @@ function RestQuery(config, auth, className, restWhere = {}, restOptions = {}, cl
...
     switch(option) {
-    case 'keys':
-      this.keys = new Set(restOptions.keys.split(','));
-        // Add the default
-      this.keys.add('objectId');
-      this.keys.add('createdAt');
-      this.keys.add('updatedAt');
+    case 'keys': {
+      const keys = restOptions.keys.split(',').concat(AlwaysSelectedKeys);
+      this.keys = Array.from(new Set(keys));
```
Comment (acinader) [inline src/RestQuery.js]: 👍

File: src/RestQuery.js [hunk 2]
```diff
@@ -52,15 +53,36 @@ function RestQuery(config, auth, className, restWhere = {}, restOptions = {}, cl
...
+
   for (var option in restOptions) {
     switch(option) {
-    case 'keys':
-      this.keys = new Set(restOptions.keys.split(','));
-        // Add the default
-      this.keys.add('objectId');
-      this.keys.add('createdAt');
-      this.keys.add('updatedAt');
+    case 'keys': {
```
Comment (acinader) [inline src/RestQuery.js]: hm. `{` in case, new one on me.

File: spec/ParseQuery.spec.js [hunk 3]
```diff
@@ -2655,6 +2655,46 @@ describe('Parse.Query testing', () => {
...
+        } else {
+          fail('bazoo should be set');
+        }
+      } else {
+        fail('barBaz should be set');
+      }
+      done();
+    })
+  });
+
```
Comment (acinader) [inline spec/ParseQuery.spec.js]: solid test.

File: src/RestQuery.js (hunk 2 above)
Comment (flovilmart) [inline src/RestQuery.js]: yeah to create a scope:  http://eslint.org/docs/rules/no-case-declarations

**Pull request merged**